vmm/src/memory_manager.rs

```diff
@@ -1846,6 +1852,88 @@ impl MemoryManager {
     pub fn acpi_address(&self) -> Option<GuestAddress> {
         self.acpi_address
     }
+
+    pub fn num_guest_ram_mappings(&self) -> u32 {
+        self.guest_ram_mappings.len() as u32
+    }
+
+    #[cfg(feature = "guest_debug")]
+    pub fn coredump_memory_regions(&self, mem_offset: u64) -> CoredumpMemoryRegions {
+        let mut mem_blocks = BTreeMap::new();
+        let mut mem_offset_in_elf = mem_offset;
+        for guest_ram_mapping in &self.guest_ram_mappings {
+            let load_info = CoredumpMemoryRegion {
+                mem_offset_in_elf: 0,
+                mem_size: guest_ram_mapping.size,
+            };
+            mem_blocks.insert(guest_ram_mapping.gpa, load_info);
+        }
+
+        let mut ram_maps = BTreeMap::new();
+        for (gpa, load) in &mem_blocks {
+            let load_info = CoredumpMemoryRegion {
+                mem_offset_in_elf,
+                mem_size: load.mem_size,
+            };
+
+            ram_maps.insert(*gpa, load_info);
+            mem_offset_in_elf += load.mem_size;
+        }
```

Comment:
Ok I understand that you need the list to be sorted by GPA. But that's something that's not obvious, which is why I suggest the following code (naming the variables correctly):
```rust
let mut mappings_sorted_by_gpa = self.guest_ram_mappings.clone();
mappings_sorted_by_gpa.sort_by_key(|m| m.gpa);

let mut mem_offset_in_elf = mem_offset;
let mut ram_maps = BTreeMap::new();
for mapping in mappings_sorted_by_gpa.iter() {
    ram_maps.insert(mapping.gpa, CoredumpMemoryRegion {
        mem_offset_in_elf,
        mem_size: mapping.size,
    });
    mem_offset_in_elf += mapping.size;
}
```

Reply:
Nice implementation! Thanks.